### PR TITLE
Aligning with changes in BHoMAdapter Refactoring Level 04

### LINKED ISSUE #81: Mongo_Toolkit: Align to changes for BHoM_Adapter refactoring level 04
#### Parent issue(s) or Pull Request(s) causing the changes:

https://github.com/BHoM/BHoM_Adapter/pull/164

#### Description:
Updating toolkit to align with refactoring of the parent BHoMAdapter.

### PR DESCRIPTION
**Pull request by alelom**
### NOTE: Depends on 
https://github.com/BHoM/BHoM_Engine/pull/1371
https://github.com/BHoM/BHoM_Adapter/pull/164
https://github.com/BHoM/BHoM_UI/pull/167

- If you use Grasshopper: https://github.com/BHoM/Grasshopper_Toolkit/pull/440
- If you use Excel: https://github.com/BHoM/Excel_Toolkit/pull/172
- If you use Dynamo: https://github.com/BHoM/Dynamo_Toolkit/pull/188

- If your Toolkit needs Socket_Toolkit: https://github.com/BHoM/Socket_Toolkit/pull/57

- If you want the Adapter components to be automatically upgraded to their new version in your test scripts: https://github.com/BHoM/Versioning_Toolkit/pull/8

 ### Issues addressed by this PR

 Closes #81

### CODE REVIEW

File: Mongo_Adapter/AdapterActions/Push.cs [hunk 1]
```diff
@@ -36,20 +38,19 @@ public partial class MongoAdapter
...
             bool replace = true;
-            if (config != null)
+            MongoConfig mongoConfig = actionConfig as MongoConfig;
+            if (actionConfig != null)
             {
-                if (config.ContainsKey("Tag"))
-                    tag = config["Tag"] as string;
-                if (config.ContainsKey("Replace"))
-                    bool.TryParse(config["Replace"] as string, out replace);
+                tag = mongoConfig.Tag;
```
Comment (alelom) [inline Mongo_Adapter/AdapterActions/Push.cs]: I preserved this

File: Mongo_oM/MongoConfig.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+using System;
+using System.ComponentModel;
+
+namespace BH.oM.Adapter.Mongo
+{
+    public class MongoConfig : ActionConfig
+    {
+        [Description("Replace the content while actioning Execute.")]
+        public bool Replace { get; set; } = false;
+        public string Tag { get; set; } = "";
```
Comment (alelom) [inline Mongo_oM/MongoConfig.cs]: Preserved here

File: Mongo_Adapter/AdapterActions/Push.cs [hunk 3]
```diff
@@ -36,20 +38,19 @@ public partial class MongoAdapter
...
+                return new List<object>();
 
             // Get the config
             bool replace = true;
-            if (config != null)
+            MongoConfig mongoConfig = actionConfig as MongoConfig;
+            if (actionConfig != null)
             {
-                if (config.ContainsKey("Tag"))
-                    tag = config["Tag"] as string;
```
Comment (alelom) [inline Mongo_Adapter/AdapterActions/Push.cs]: @adecler Was this actually needed? Why can't this work with the adapter input `tag`?
As you can see I preserved this [here](https://github.com/BHoM/Mongo_Toolkit/pull/82/files#r354815637) and [here](https://github.com/BHoM/Mongo_Toolkit/pull/82/files#r354815882), but I wonder if we couldn't remove it altogether.

File: Mongo_oM/Mongo_oM.csproj [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{62D0B8AE-646C-46F1-8C28-1AED1ED73D54}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Mongo_oM1</RootNamespace>
+    <AssemblyName>Mongo_oM1</AssemblyName>
```
Comment (FraserGreenroyd) [inline Mongo_oM/Mongo_oM.csproj]: Assembly name and root namespace shouldn't be `Mongo_oM1` - it should be `Mongo_oM` surely?

**Pull request merged**